src/main/kotlin/stringaddcalculator/domain/Number.kt

```diff
@@ -0,0 +1,40 @@
+package stringaddcalculator.domain
+
+class Number {
+    val value: Int
+
+    constructor(number: String) {
+        try {
+            this.value = number.toInt()
+        } catch (e: Exception) {
+            throw IllegalArgumentException("$number 는 숫자가 아닙니다.")
+        }
+    }
+
+    constructor(number: Int) {
+        this.value = number
+    }
+
+    fun plus(number: Number): Number {
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
    operator fun plus(number: Number): Number {
```